contrib/internal/httptrace/httptrace.go

```diff
@@ -41,6 +61,14 @@ func StartRequestSpan(r *http.Request, opts ...ddtrace.StartSpanOption) (tracer.
 	if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(r.Header)); err == nil {
 		opts = append(opts, tracer.ChildOf(spanctx))
 	}
+	if enabled, err := collectClientIP(); err != nil {
```

Comment:
NIT: I think the usual pattern is to lookup and parse env based config vars in the config and to just check the config in places like this. That keeps the code a little more organized and also has performance benefits.

https://github.com/DataDog/dd-trace-go/blob/74a8e95a31d9fc95d2f1a7b804fabe7f2467cb0c/contrib/internal/httptrace/config.go#L28-L49

Reply:
Indeed, it is important doing the initializations and configurations out of the "hot path" when possible.